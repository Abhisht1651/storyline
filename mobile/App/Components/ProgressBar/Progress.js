// @flow

import React from 'react'
import { View } from 'react-native'
import ProgressBar from './ProgressBar'
import styles from '../Styles/ProgreeBarStyles/ProgressStyle'

type Props = {
  count: Number
}

const Progress = (props: Props) => {
  const { count } = props
  let progress1 = 'unfilled'
  let progress2 = 'unfilled'
  let progress3 = 'unfilled'
  if (count === 1) {
    progress1 = 'filled'
  } else if (count === 2) {
    progress1 = 'filled'
    progress2 = 'filled'
  } else if (count === 3) {
    progress1 = 'filled'
    progress2 = 'filled'
    progress3 = 'filled'
  }
  return (
    <View style={styles.container}>
      <View style={styles.progressBarContainer}>
        <ProgressBar styleType={progress1} />
        <ProgressBar styleType={progress2} />
        <ProgressBar styleType={progress3} />
      </View>
    </View>
  )
}

export default Progress
