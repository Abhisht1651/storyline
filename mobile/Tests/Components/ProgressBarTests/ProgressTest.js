import { shallow } from 'enzyme'
import React from 'react'
import Progress from '../../../App/Components/ProgressBar/Progress'

describe('Progress', () => {
  it('should match to snapshot', () => {
    const component = shallow(<Progress />)
    expect(component).toMatchSnapshot()
  })

  it('should render ProgressBar', () => {
    const component = shallow(<Progress />)
    expect(component.find('ProgressBar')).toHaveLength(3)
  })
})
