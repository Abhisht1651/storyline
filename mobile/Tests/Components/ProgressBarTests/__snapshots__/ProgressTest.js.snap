// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress should match to snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 40,
      }
    }
  >
    <ProgressBar
      styleType="unfilled"
    />
    <ProgressBar
      styleType="unfilled"
    />
    <ProgressBar
      styleType="unfilled"
    />
  </View>
</View>
`;
